test/suites/upgrade-ostree.robot

```diff
@@ -0,0 +1,85 @@
+*** Settings ***
+Documentation       Tests related to MicroShift upgradeability on OSTree-based systems
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/microshift-process.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags          ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+
+*** Test Cases ***
+Rebooting Healthy System Should Result In Data Backup
+    [Documentation]    Check if rebooting healthy system will result in backing up of MicroShift data
+
+    Wait Until Greenboot Health Check Exited
+    System Should Be Healthy
+    Remove Existing Backup For Current Deployment
+
+    Reboot MicroShift Host
+    Wait Until MicroShift Service Is Active
+
+    Backup For Booted Deployment Should Exist
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+
+Teardown
+    [Documentation]    Test suite teardown
+    Logout MicroShift Host
+
+Wait Until Greenboot Health Check Exited
+    [Documentation]    Wait until greenboot healthchecks are done
+
+    Wait Until Keyword Succeeds    5m    15s
+    ...    Greenboot Health Check Exited
+
+Wait Until MicroShift Service Is Active
+    [Documentation]    Wait until MicroShift service is ready
+    Wait Until Keyword Succeeds    5m    15s
+    ...    MicroShift Service Is Active
+
+Greenboot Health Check Exited
+    [Documentation]    Checks if greenboot-healthcheck finished running
+
+    ${value}=    Get Systemd Setting    greenboot-healthcheck.service    SubState
+    Should Be Equal As Strings    ${value}    exited
+
+Remove Existing Backup For Current Deployment
+    [Documentation]    If exists, remove backup for current deployment
+
+    ${exists}=    Does Backup For Booted Deployment Exist
+    IF    ${exists} == ${False}    RETURN
+
+    ${deployID}=    Get Booted Deployment ID
+
+    ${rm_output}    ${rm_stderr}    ${rc}=    Execute Command
+    ...    find ${BACKUP_STORAGE} -name '${deployID}*' -exec rm -rf {} +
```

Comment:
We know the exact path, right? Can we just remove it, instead of using `find`?

Reply:
OK, we probably want to do something to limit the depth of the find call, then. Is it enough to say `rm -rf ${deployID}*` instead of using find at all?

And we probably want to check for deployID being empty, to ensure we don't delete everything.

Reply:
`find` was only looking into `/var/lib/microshift-backups/` but I changed it to `rm -rf` because it's easier to read